### PR TITLE
Fix ariaLabel on nameless products

### LINKED ISSUE #35783: Product grid is not getting displayed, getting undefined array key "name" in logs.
### Preconditions and environment

- Magento 2.4.4
- Upgraded from M2.4.1
- PHP 8.1

### Steps to reproduce

1. Login as admin.
2. Go to Store->attributes->product search code "name" and set Values Required to "NO"
3. Create a product without name in 2.4.1 and then upgrade to 2.4.4 along with PHP.
4. Now, create a Product without name after setting the 'Name' attribute as not mandatory.

### Expected result

The Product Grid should load fine as usual.

### Actual result

The Product Grid throws an alert of something went wrong and freezes.

### Additional information

The Log shows the below error:
`main.CRITICAL: Exception: Warning: Undefined array key "name" in /vendor/magento/module-catalog/Ui/Component/Listing/Columns/ProductActions.php on line 61 in /vendor/magento/framework/App/ErrorHandler.php:61`

### Release note

_No response_

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by vandijkstef**
### Description (*)

Safely renders the ariaLabel on the product admin grid should a product not have a name set so the page stays functional

### Fixed Issues (if relevant)

1. Fixes magento/magento2#35783

### Manual testing scenarios (*)

1. Create a product without a name (aka without an entry in the catalog_product_entity_varchar on the name attribute)
2. Open the admin panel -> Catalog -> Products and make sure that product is included in the view
3. It should error/free

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by vandijkstef** [comment]
@magento run all tests

**Comment by vandijkstef** [comment]
Don't think the failed tests are due to the changes

File: app/code/Magento/Catalog/Ui/Component/Listing/Columns/ProductActions.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function prepareDataSource(array $dataSource)
                         'catalog/product/edit',
                         ['id' => $item['entity_id'], 'store' => $storeId]
                     ),
-                    'ariaLabel' => __('Edit ') . $item['name'],
+                    'ariaLabel' => __('Edit ') . (isset($item['name']) ? $item['name'] : ''),
```
Comment (MeCapron) [inline app/code/Magento/Catalog/Ui/Component/Listing/Columns/ProductActions.php]: Parenthesis are meaningless here.

You could simplify using : 

```suggestion
                    'ariaLabel' => __('Edit ') . $item['name'] ?? '',
```

A better approach could probably be to define it directly in the translation, but the translation needs to be add

```php
                    'ariaLabel' => __('Edit %1', $item['name'] ?? ''),
```

Comment (vandijkstef) [reply app/code/Magento/Catalog/Ui/Component/Listing/Columns/ProductActions.php]: Thanks, will have a look later this week

Comment (vandijkstef) [inline app/code/Magento/Catalog/Ui/Component/Listing/Columns/ProductActions.php]:
```suggestion
                    'ariaLabel' => __('Edit') . ' ' . ($item['name'] ?? ''),
```

- Thnx, wasn't aware nullish coalescing was a thing in PHP now
- The translation wasn't working anyway due to the added space at the end
- The parenthesis are absolutely NOT meaningless here, they enforce evaluation, without it, it still breaks for me (on PHP 7.4)
- I do absolutely agree that including it in the translation is best, but the issue at hand is possibly breaking admin views, and any other translations needs to be updated as well. I've checked if the pattern was accidentaly available but it is not

**Comment by vandijkstef** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Ui/Component/Listing/Columns/ProductActions.php (hunk 1 above)
Comment (vandijkstef) [reply app/code/Magento/Catalog/Ui/Component/Listing/Columns/ProductActions.php]: @MeCapron Could you take a look at this, or tell me what I need to do to move this forward?

**Comment by HenKun** [comment]
@vandijkstef Do you have any idea how to continue here? Seems pretty simple and correct fix

**Comment by vandijkstef** [comment]
@HenKun Nope it's been stale for quite a while and I've given up on chasing it 🤷‍♂️

**Comment by hostep** [comment]
Can everyone who is interested in this fix give the PR a thumbs up? The more thumbs-ups a PR has, [the more likely it will be picked up](https://community.magento.com/t5/Magento-DevBlog/Get-involved-with-the-Magento-Open-Source-Community/ba-p/503497) for being moved forward.

**Pull request closed without merging**